rio_tiler/io/cogeo.py

```diff
@@ -198,10 +195,24 @@ def _set_zooms(self):
             )
             minzoom, maxzoom = self.tms.minzoom, self.tms.maxzoom
 
-        self.minzoom = self.minzoom if self.minzoom is not None else minzoom
-        self.maxzoom = self.maxzoom if self.maxzoom is not None else maxzoom
+        self._minzoom = minzoom
+        self._maxzoom = maxzoom
         return
 
+    @property
+    def minzoom(self):
```

Comment:
It's sort of weird to me that calling `minzoom` or `maxzoom` sets both.  Its probably not a huge deal but it is a side effect that I think we should try to avoid if possible.  Maybe there is a clever way to split the zoom methods into two, where one handles min and the other handles max.

I wonder how expensive `calculate_default_transform` is.  If it's cheap it might make sense to just call it twice, once for min and once for max.  Then we could break this up.

Reply:
Good point, 
`calculate_default_transform` can be expensive but `min/max zoom` level shouldn't be called too often so I think we can split the `get_zooms`. I'll work on this tomorrow!